bgpd/bgp_keepalives.c

```diff
@@ -175,6 +175,15 @@ void *bgp_keepalives_start(void *arg)
 	struct timeval next_update = {0, 0};
 	struct timespec next_update_ts = {0, 0};
 
+	/*
+	 * The RCU mechanism for each pthread is initialized in a "locked"
+	 * state. That's ok for pthreads using the frr_pthread,
+	 * thread_fetch event loop, because that event loop unlocks regularly.
+	 * For foreign pthreads, the lock needs to be unlocked so that the
+	 * background rcu pthread can run.
+	 */
+	rcu_read_unlock();
```

Comment:
This thread is not using any logging functions after this point? Otherwise it'd need to reaquire and release the lock around that…

Reply:
> I think the key zlog functions - the buffer-flush functions and the per-target logfn - have rcu locks around them.

Right, not sure why I thought they didn't. Not enough caffeine or something.